client/src/api.js

```diff
@@ -55,7 +55,8 @@ module.exports.signin = function (email, password) {
     // TODO: handle bad creds properly
     return post('/login', {email, password})
         .then( (response) => {
-           return response && response.user
+            // I don't think this code is reachable
```

Comment:
why not
it chains another promise to the one returned by response.json() in line 15.
Whether or not it's the right way to go - that's a different question ...